src/js/integrations/spotlight.ts

```diff
@@ -0,0 +1,96 @@
+import type { Client, Envelope, EventProcessor, Integration } from '@sentry/types';
+import { logger, serializeEnvelope } from '@sentry/utils';
+
+import { makeUtf8TextEncoder } from '../transports/TextEncoder';
+import { ReactNativeLibraries } from '../utils/rnlibraries';
+
+type SpotlightReactNativeIntegrationOptions = {
+  /**
+   * The URL of the Sidecar instance to connect and forward events to.
+   * If not set, Spotlight will try to connect to the Sidecar running on localhost:8969.
+   *
+   * @default "http://localhost:8969/stream"
+   */
+  sidecarUrl?: string;
+};
+
+/**
+ *
+ */
+export function Spotlight({
+  sidecarUrl = getDefaultSidecarUrl(),
+}: SpotlightReactNativeIntegrationOptions = {}): Integration {
+  logger.info('[Spotlight] Using Sidecar URL', sidecarUrl);
+
+  return {
+    name: 'Spotlight',
+
+    setupOnce(_: (callback: EventProcessor) => void, getCurrentHub) {
+      const client = getCurrentHub().getClient();
+      if (client) {
+        setup(client, sidecarUrl);
+      } else {
+        logger.warn('[Spotlight] Could not initialize Sidecar integration due to missing Client');
+      }
+    },
+  };
+}
+
+function setup(client: Client, sidecarUrl: string): void {
+  sendEnvelopesToSidecar(client, sidecarUrl);
+}
+
+function sendEnvelopesToSidecar(client: Client, sidecarUrl: string): void {
+  if (!client.on) {
+    return;
+  }
+
+  client.on('beforeEnvelope', (originalEnvelope: Envelope) => {
+    // TODO: This is a workaround for spotlight/sidecar not supporting images
+    const spotlightEnvelope: Envelope = [...originalEnvelope];
+    const envelopeItems = [...originalEnvelope[1]].filter(
+      item => typeof item[0].content_type !== 'string' || !item[0].content_type.startsWith('image'),
+    );
```

Comment:
Hmm this is a good point. I think we just support text data at the moment.

I think we should handle this in the sidecar somehow because we don't want SDKs to have too much of that logic. Ideally, Spotlight (similarly to Relay lol) accepts all kinds of payloads and just deals with removing incompatible stuff on its end. 

Additionally, WDYT, would it be valuable to render the images in the Spotlight UI?

Reply:
Btw, no change required from my end. The workaround is fine for now, I just think we should handle this in the spotlight sidecar in the future

Reply:
I think rendering the screenshots would be especially useful for crashes/uncaught errors. So the developer doesn't have to try to crash the app again and carefully watch what was on the screen at that moment (or create a screenrecording).